atkinson/dlrn/http_data.py

```diff
@@ -69,81 +77,97 @@ def dlrn_http_factory(host, config_file=None, logger=getLogger()):
+    if link_name is not None:
+        link = link_name
+
     return DlrnHttpData(config[host]['url'],
                         config[host]['release'],
+                        link_name=link,
                         logger=logger)
 
 
 class DlrnHttpData():
     """A class used to interact with the dlrn API"""
-    def __init__(self, url, release, logger=getLogger()):
+    def __init__(self, url, release, link_name='current', logger=getLogger()):
         """
         Class constructor
 
         :param url: The URL to the host to obtain data.
         :param releases: The release name to use for lookup.
+        :param link_name: The name of the dlrn symlink to fetch data from.
         :param logger: An atkinson logger to use. Default is the base logger.
         """
-        self.url = url
+        self.url = os.path.join(url, release)
         self.release = release
         self._logger = logger
+        self._link_name = link_name
+        self._commit_data = {}
+        self._fetch_commit()
+
+    def _fetch_commit(self):
+        """
+        Fetch the commit data from dlrn
+        """
+        full_url = os.path.join(self.url,
+                                self._link_name,
+                                'commit.yaml')
+        data = _fetch_yaml(full_url, self._logger)
+        if data is not None and 'commits' in data:
+            pkg = data['commits'][0]
+            if pkg['status'] == 'SUCCESS':
+                self._commit_data = {'name': pkg['project_name'],
+                                     'dist_hash': pkg['distro_hash'],
+                                     'commit_hash': pkg['commit_hash']}
+                if 'extended_hash' in pkg:
+                    self._commit_data['extended_hash'] = pkg['extended_hash']
+                else:
+                    self._commit_data['extended_hash'] = 'None'
+            else:
+                msg = '{0} has a status of error'.format(str(pkg))
+                self._logger.warning(msg)
 
-    def _build_url(self, commit_hash, distgit_hash):
+    def _build_url(self):
         """
         Generate a url given a commit hash and distgit hash to match the format
         base/AB/CD/ABCD123_XYZ987 where ABCD123 is the commit hash and XYZ987
         is a portion of the distgit hash.
 
-        :param commit_hash: A string with the full commit hash on the change
-        :param dlrn_hash: A string with the full dlrn hash to fetch from
         :return: A string with the full URL.
         """
-        first = commit_hash[0:2]
-        second = commit_hash[2:4]
-        third = commit_hash + '_' + distgit_hash[0:8]
-        return os.path.join(self.url, self.release, first, second,
+        first = self._commit_data['commit_hash'][0:2]
+        second = self._commit_data['commit_hash'][2:4]
+        third = self._commit_data['commit_hash'] + '_'
+        third += self._commit_data['dist_hash'][0:8]
+        if self._commit_data['extended_hash'] != 'None':
```

Comment:
If you were to take my suggestion above, you would want to either change this to check against None instead of the string, or simplify it to:
```
if self._commit_data.get('extended_hash'):
```

Note also, it might be more consistent to change all calls to the dict(s) to use get() for consistency, plus you get the benefit of not throwing an error if a key is missing (unless you want the error, I guess).